util/collate/collate.go

```diff
@@ -217,7 +218,7 @@ func SubstituteMissingCollationToDefault(co string) string {
 	if _, err = GetCollationByName(co); err == nil {
 		return co
 	}
-	logutil.BgLogger().Warn(err.Error())
+	logutil.BgLogger().Warn(fmt.Sprintf("The collation %s specified on connection is not supported when new collation is enabled, switch to the default collation: %s", co, mysql.DefaultCollationName))
```

Comment:
Doesn't this generate a lot of log spam? Imagine a scenario where a user uses a MySQL 8 connector, we don't support the default collation - I assume this will mean one log-line per connect?

Reply:
yes, one log-line per connect. I think it is ok because it is the same as in the past.